src/Utils/Match.ts

```diff
@@ -0,0 +1,26 @@
+export class UtilMatch {
+
+     public static mapToClosingChar =  {"'"  : "'",
+                                        "\"" : "\"", 
+                                        "<"  : ">",
+                                        ">"  : ">",
+                                        "("  : ")",
+                                        ")"  : ")",
+                                        "["  : "]",
+                                        "]"  : "]",
+                                        "{"  : "}", 
+                                        "}"  : "}",
+                                        };
```

Comment:
@aioutecism 

``` typescript
const openingToClosing: {[key: string]: string} = {
    '\'' : '\'',
    '"'  : '"',
    '<'  : '>',
    '>'  : '>',
    '('  : ')',
    ')'  : ')',
    '['  : ']',
    ']'  : ']',
    '{'  : '}',
    '}'  : '}',
};
const closingToOpening: {[key: string]: string} = {};
Object.keys(openingToClosing).forEach(opening => {
    closingToOpening[openingToClosing[opening]] = opening;
});
```

This won't do what we want, since `openingToClosing` maps both "(" and ")" to ")", for example.  `closingToOpening` should map both to "(".